retriever.py

```diff
@@ -245,29 +247,77 @@ def main():
             pres_min = mslib.msui.mpl_qtwidget.MplSideViewCanvas._pres_min
-            major_ticks = pres_maj[(pres_maj <= p_bot) & (pres_maj >= p_top)]
-            minor_ticks = pres_min[(pres_min <= p_bot) & (pres_min >= p_top)]
-            labels = ["{}".format(int(l / 100.))
-                      if (l / 100.) - int(l / 100.) == 0 else "{}".format(float(l / 100.)) for l in major_ticks]
-            if len(labels) > 20:
-                labels = ["" if x.split(".")[-1][0] in "975" else x for x in labels]
-            elif len(labels) > 10:
-                labels = ["" if x.split(".")[-1][0] in "9" else x for x in labels]
-            ax.set_ylabel("pressure (hPa)")
+            lat_inds = np.arange(len(lats))
+            tick_index_step = len(lat_inds) // num_labels
+            type = config_loader(dataset="type")
+
+            if type == "no secondary axis":
+                major_ticks = [] * units.pascal
+                minor_ticks = [] * units.pascal
+                labels = []
+                ylabel = ""
+            elif type == "pressure":
+                # Compute the position of major and minor ticks. Major ticks are labelled.
+                major_ticks = pres_maj[(pres_maj <= p_bot) & (pres_maj >= p_top)]
+                minor_ticks = pres_min[(pres_min <= p_bot) & (pres_min >= p_top)]
+                labels = [f"{int(_x / 100)}"
+                        if (_x / 100) - int(_x / 100) == 0 else f"{float(_x / 100)}" for _x in major_ticks]
+                if len(labels) > 20:
+                    labels = ["" if x.split(".")[-1][0] in "975" else x for x in labels]
+                elif len(labels) > 10:
+                    labels = ["" if x.split(".")[-1][0] in "9" else x for x in labels]
+                ylabel = "pressure (hPa)"
+            elif type == "pressure altitude":
+                bot_km = thermolib.pressure2flightlevel(p_bot * units.Pa).to(units.km).magnitude
+                top_km = thermolib.pressure2flightlevel(p_top * units.Pa).to(units.km).magnitude
+                ma_dist, mi_dist = 4, 1.0
+                if (top_km - bot_km) <= 20:
+                    ma_dist, mi_dist = 1, 0.5
+                elif (top_km - bot_km) <= 40:
+                    ma_dist, mi_dist = 2, 0.5
+                major_heights = np.arange(0, top_km + 1, ma_dist)
+                minor_heights = np.arange(0, top_km + 1, mi_dist)
+                major_ticks = thermolib.flightlevel2pressure(major_heights * units.km).magnitude
+                minor_ticks = thermolib.flightlevel2pressure(minor_heights * units.km).magnitude
+                labels = major_heights
+                ylabel = "pressure altitude (km)"
+            elif type == "flight level":
+                bot_km = thermolib.pressure2flightlevel(p_bot * units.Pa).to(units.km).magnitude
+                top_km = thermolib.pressure2flightlevel(p_top * units.Pa).to(units.km).magnitude
+                ma_dist, mi_dist = 50, 10
```

Comment:
the limit values seem like some constants. Maybe those should later be defined as consants too?